### PR TITLE
[Terra-form-single-select] Fix Keyboard interaction for form-single-select

### PR DESCRIPTION
**Pull request by gt106551**
### **Keyboard Navigation**
 Pressed Tab to navigate to Single Select in first example and Pressed Down Arrow to expand the list box.
 Navigated the list of options using the Down and Up Arrow keys and Pressed Enter Space to select the option. Selection did not work with Space.
### Screen reader
Pressed Tab to navigate to Single Select  example and Pressed Down Arrow to expanded the single select example and  Enter did not work to expand the single select example 
**What was changed:**
 Added  KeyCode.KEY_RETURN
Added  KeyCode.KEY_SPACE
**Why it was changed:**

Selection did not work with Space key.
 Enter did not work to expand the list box

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

[UXPLATFORM-9628](https://jira2.cerner.com/browse/UXPLATFORM-9628)

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-form-select/src/single/Menu.jsx [hunk 1]
```diff
@@ -193,7 +193,7 @@ class Menu extends React.Component {
       this.scrollTimeout = setTimeout(this.clearScrollTimeout, 500);
       this.setState({ active: MenuUtil.findNext(children, active) }, this.scrollIntoView);
       this.updateCurrentActiveScreenReader();
-    } else if (keyCode === KeyCode.KEY_RETURN && active !== null) {
+    } else if ((keyCode === KeyCode.KEY_RETURN || keyCode === KeyCode.KEY_SPACE) && active !== null) {
```
Comment (sdadn) [inline packages/terra-form-select/src/single/Menu.jsx:196]: there needs to be a test for space

File: packages/terra-form-select/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 
 ## Unreleased
+* Fixed
+  * Fixed selection did not work with space-key `form-single-select`.
```
Comment (sdadn) [inline packages/terra-form-select/CHANGELOG.md:5]:
```suggestion
  * Fixed selection not working with space key in `form-single-select`.
```

File: packages/terra-form-select/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
...
+  * Fixed Enter did not work to expand the combo box example `form-single-select` .
```
Comment (sdadn) [inline packages/terra-form-select/CHANGELOG.md:6]:
```suggestion
  * Fixed enter key not expanding the combo box in `form-single-select` .
```

**Review by sdadn — CHANGES_REQUESTED** [review]

File: packages/terra-form-select/src/single/Menu.jsx (hunk 1 above)
Comment (gt106551) [reply packages/terra-form-select/src/single/Menu.jsx:196]: Added a test case for this

File: packages/terra-form-select/CHANGELOG.md (hunk 2 above)
Comment (gt106551) [reply packages/terra-form-select/CHANGELOG.md]: updated

File: packages/terra-form-select/CHANGELOG.md (hunk 3 above)
Comment (gt106551) [reply packages/terra-form-select/CHANGELOG.md]: updated

File: packages/terra-form-select/tests/wdio/select-spec.js [hunk 4]
```diff
@@ -134,6 +134,7 @@ Terra.describeViewports('Select', ['tiny'], () => {
       });
 
       it('should display closed dropdown', () => {
+        browser.keys('Enter');
```
Comment (supreethmr) [inline packages/terra-form-select/tests/wdio/select-spec.js:137]: why do we need an enter to close the dropdown

File: packages/terra-form-select/CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 
 ## Unreleased
+* Fixed
+   * Fixed selection not working with space key in `form-single-select`.
+  * Fixed enter key not expanding the combo box in `form-single-select` .
```
Comment (supreethmr) [inline packages/terra-form-select/CHANGELOG.md:6]:
```suggestion

* Fixed
   * Fixed selection not working with space key in `form-single-select`.
   * Fixed enter key not expanding the combo box in `form-single-select` .
```

File: packages/terra-form-select/tests/wdio/select-spec.js [hunk 6]
```diff
@@ -326,6 +308,28 @@ Terra.describeViewports('Select', ['tiny'], () => {
...
     });
 
+    describe('default should select an option by pressing space', () => {
+      it('default should open the dropdown by clicking the toggle', () => {
+        $('[data-terra-select]').click();
+      });
+
+      it('should display open dropdown', () => {
+        Terra.validates.element('[enter]open-dropdown', { selector: '#root' });
+      });
```
Comment (supreethmr) [inline packages/terra-form-select/tests/wdio/select-spec.js:318]: we can remove this block !!

Comment (supreethmr) [inline packages/terra-form-select/tests/wdio/select-spec.js:318]:
```suggestion
```
let's remove this block

File: packages/terra-form-select/CHANGELOG.md (hunk 5 above)
Comment (gt106551) [reply packages/terra-form-select/CHANGELOG.md]: Updated.

File: packages/terra-form-select/tests/wdio/select-spec.js (hunk 6 above)
Comment (gt106551) [reply packages/terra-form-select/tests/wdio/select-spec.js]: updated.

**Review by supreethmr — APPROVED** [review]

**Review by saket2403 — APPROVED** [review]

**Pull request merged**